packages/filebrowser-extension/src/index.ts

```diff
@@ -903,6 +904,140 @@ const openUrlPlugin: JupyterFrontEndPlugin<void> = {
   }
 };
 
+const notifyUploadPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/filebrowser-extension:notify-upload',
+  requires: [IDefaultFileBrowser, ISettingRegistry, ITranslator],
+  autoStart: true,
+  activate: async (
+    app: JupyterFrontEnd,
+    defaultBrowser: FileBrowser,
+    settingRegistry: ISettingRegistry,
+    translator: ITranslator
+  ) => {
+    const trans = translator.load('jupyterlab');
+    // load and watch settings
+    const settings = await settingRegistry.load(FILE_BROWSER_PLUGIN_ID);
+    let autoOpen = settings.get('autoOpenUploads').composite as boolean;
+    let maxSize =
+      (settings.get('maxAutoOpenSizeMB').composite as number) * 1024 * 1024;
+
+    settings.changed.connect(() => {
+      autoOpen = settings.get('autoOpenUploads').composite as boolean;
+      maxSize =
+        (settings.get('maxAutoOpenSizeMB').composite as number) * 1024 * 1024;
+    });
+
+    // attach to the Uploader after restore
+    void app.restored.then(() => {
+      const widgets = Array.from(defaultBrowser.toolbar.children());
+      const uploader = widgets.find(w => w instanceof Uploader) as
+        | Uploader
+        | undefined;
+      if (!uploader) {
+        console.warn('Uploader widget not found');
+        return;
+      }
+
+      uploader.filesUploaded.connect((_sender, models) => {
+        // single-file case
+        // Get all allowed extensions from default file types
+        const defaultFileTypes = DocumentRegistry.getDefaultFileTypes();
+        const allowedExtensions = defaultFileTypes.reduce<string[]>(
+          (acc, ft) => {
+            if (ft.extensions) {
+              acc.push(...ft.extensions);
+            }
+            return acc;
+          },
+          []
+        );
+        // Check if the uploaded file has an allowed extension
+        const fileExt = models[0].name
+          .toLowerCase()
+          .slice(models[0].name.lastIndexOf('.'));
+        const isAllowedFileType = allowedExtensions.includes(fileExt);
```

Comment:
Instead of `lastIndexOf` we may need to loop over them and check suffixes because the extensions may be composite, e.g. `.table.json` as per https://jupyterlab.readthedocs.io/en/4.1.x/api/interfaces/docregistry.DocumentRegistry.IFileType.html#extensions